.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+
+  build_kernel:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y build-essential gawk flex bison openssl dkms \
+          libelf-dev libudev-dev libpci-dev libiberty-dev autoconf dwarves \
+          libncurses-dev libssl-dev
+
+      - name: Generate .config 
+        run: |
+          rm -rf .git .gitattributes .gitignore
```

Comment:
Why is this necessary?

Reply:
Kernel makefile tries to interact with git for determining version, which results in mis-configured version and the build breaks due to that.

Reply:
I did hit the build problems myself and had to rename .git directory there and back before and after each build to still use git in between.  It is just a waste to re-download full repo for every CI run. Some sources propose to use `git clone --depth 1` at least (unless it is done already here).